### PR TITLE
Reformating lars_update #2334

### PR DESCRIPTION
**Pull request by vandit98**

### CODE REVIEW

**Comment by jkeane508** [comment]
Hi @vandit98 can you let me know which issue you are creating this PR for?

**Comment by vandit98** [comment]
i have created this for lars_update #2334 in gradient.py for array function reformatting.
[](https://github.com/unifyai/ivy/issues/2334)

**Comment by jkeane508** [comment]
Hi @vandit98,

You are missing docstring examples for the functional and container implementations of this function. You can refer to the documentation [here](https://lets-unify.ai/ivy/deep_dive/13_docstring_examples.html) on how to complete this task.

Thanks,
James

**Review by jkeane508 — CHANGES_REQUESTED** [review]
As mentioned above

**Comment by vandit98** [comment]
thanks for mentioning i will update it.

**Comment by vandit98** [comment]
i have updated as per requirement.

**Comment by vandit98** [comment]
requesting for re-review.

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hi, 

You still need to add examples to the implementation in the functional module [here](https://github.com/unifyai/ivy/blob/9088011295737beacc7ed2d64929ec7ccd1428dc/ivy/functional/ivy/gradients.py#L778)

Thanks!

**Comment by vandit98** [comment]
hi
i have added the example in the functional module as well.

**Comment by jkeane508** [comment]
@vandit98 There seems to be some conflicts with the gradients file, if you could resolve to allow tests to run. If the docstring tests are passing and there are no lint errors this is good to merge!

File: ivy/array/gradients.py [hunk 1]
```diff
@@ -138,19 +138,19 @@ def gradient_descent_update(
 
         Parameters
         ----------
-        self
+        w
```
Comment (jkeane508) [inline ivy/array/gradients.py:141]: As this is an instance method this should be self

File: ivy/container/gradients.py [hunk 2]
```diff
@@ -654,6 +654,70 @@ def static_lars_update(
         prune_unapplied: bool = False,
         map_sequences: bool = False,
     ) -> ivy.Container:
+        """
+        ivy.Container instance method variant of ivy.lars_update.
```
Comment (jkeane508) [inline ivy/container/gradients.py:658]: This is incorrect as this is the static implementation of this function

File: ivy/container/gradients.py [hunk 3]
```diff
@@ -654,6 +654,70 @@ def static_lars_update(
...
+        This method simply wraps the function, and so the docstring for
+        ivy.lars_update also applies to this method
+        with minimal changes.
+        Parameters
+        ----------
+        self
```
Comment (jkeane508) [inline ivy/container/gradients.py:664]: As this is a static method it does not accept self as an argument

File: ivy/container/gradients.py [hunk 4]
```diff
@@ -654,6 +654,70 @@ def static_lars_update(
...
+            The new weights , following the lars updates.
+        Examples
+        --------
+        With :code: `ivy.container` inputs:
+        >>> w = ivy.Container(a=ivy.array([1., 2., 3.]),\
+                              b=ivy.array([3.48, 5.72, 1.98]))
+        >>> dcdw = ivy.Container(a=ivy.array([0.5, 0.2, 0.1]),\
+                                 b=ivy.array([2., 3.42, 1.69]))
+        >>> lr = ivy.array(0.3)
+        >>> NewWeights = w.lars_update(dcdw, lr, inplace=False)
```
Comment (jkeane508) [inline ivy/container/gradients.py:702]: This example should use static_lars_update() not the instance method below

File: ivy/functional/ivy/gradients.py [hunk 5]
```diff
@@ -812,6 +812,17 @@ def lars_update(
     -------
     ret
         The new function weights ws_new, following the LARS updates.
+     Examples
+        --------
+        With :code:`ivy.Array` inputs:
+        >>> w = ivy.array([[[5., 3., 2.], [0., 4., 1.], [-2., 3., -1.]]])
+        >>> dcdw = ivy.array([[[0.5, 0.92, 0.1], [0.2, 0.7, 0.3], [0.3, 0.8, 0.01]]])
+        >>> lr = ivy.array(0.3)
+        >>> ws_new = w.lars_update(dcdw, lr, inplace=False)
```
Comment (jkeane508) [inline ivy/functional/ivy/gradients.py:821]: This is using the array instance method here we should use the functional implementation as outline above

**Review by jkeane508 — CHANGES_REQUESTED** [review]
I had another look through this and noticed some errors with the implementations. You can refer to docs [here](https://lets-unify.ai/ivy/deep_dive/13_docstring_examples.html#docstring-examples) on what requirements need to be met to complete this task :)

**Pull request closed without merging**